apps/system/js/app_usage_metrics.js

```diff
@@ -301,8 +310,7 @@
         }
         else {
           // Our device id does not need to be unique, just probably unique.
-          // And it doesn't even need to be a real UUID
-          self.deviceID = Math.random().toString(36).substring(2, 10);
+          self.deviceID = uuid();
           asyncStorage.setItem(DEVICE_ID_KEY, self.deviceID);
```

Comment:
You need to update DEVICE_ID_KEY to a new string to convert v2.1 devices to the new uuid format. Otherwise old devices will continue to report with a non uuid. Or you need to check that the value looks like a uuid (probably by checking that it has hyphens in it).

I'm assuming that you want to force everyone to use a uuid, even at the cost of losing continuity with the old style ids.

Or do something fancy like send oldDeviceID for devices that were upgraded.